### ISSUE TITLE
Automatically redirect to requested URL after successful SSO login

### ISSUE THREAD

**Issue by yonirab** [argoproj/argo-workflows#11000]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

This issue was originally reported and fixed in the past under https://github.com/argoproj/argo-workflows/issues/4370.
See the description there.

It worked for a couple of years, but got broken somewhere in 3.x (not sure which version exactly).

### Version

v3.4.7

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
This happens with any workflow.
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

**Comment by terrytangyuan** [comment]
Could you describe the behavior you observe? Is it redirected to an incorrect link? What's the difference in behaviors after upgrade?

**Comment by tooptoop4** [comment]
I face this. someone gives u a direct link to a specific workflow/archived workflow, but you have not logged in for the day, you enter the direct link in the browser, it takes u to login page, u login, then it just takes u to home page (workflow list) when it should really auto-redirect you to the original direct link

**Comment by yonirab** [comment]
@terrytangyuan The observed behaviour us exactly as described by @tooptoop4

**Comment by tooptoop4** [comment]
crunchy

**Comment by caelan-io** [comment]
@tooptoop4 @yonirab - could you please help us narrow down the version this might have regressed/ broken in? 3.3.X or 3.4.X? Thank you.

**Comment by sarabala1979** [comment]
I think it is not regression. it is implemented this way. instead of redirecting to the requested URL. it will redirect to the home URL.

**Comment by sarabala1979** [comment]
@tooptoop4 Here the code which is assigning redirect URL to baseurl instead of `request` url. Do you like to take a look?

https://github.com/argoproj/argo-workflows/blob/f5e31f8f36b32883087f783cb1227490bbe36bbd/server/auth/sso/sso.go#L313

**Comment by mbeers98** [comment]
I was also running into this issue. I found this happens with the `--secure=false` flag & having argo-workflows behind a reverse proxy (e.g ingress-nginx) that does TLS termination. The code linked by @sarabala1979 shows the redirect url only gets set if it matches a prefix created from: protocol, host, and baseHRef.

When running with the `--secure=false` flag the protocol will be `http` however because ingress-nginx is handling tls for my particular setup the redirect url saved in the cookie will have `https`, causing the check to fail and the redirect to default to the home page.

Hope this helps anyone else who hits this issue!

**Comment by agilgur5** [comment]
> This issue was originally reported and fixed in the past under #4370.

That was a feature request for it to exist originally, not a bug.

> I was also running into this issue. I found this happens with the `--secure=false` flag & having argo-workflows behind a reverse proxy (e.g ingress-nginx) that does TLS termination.

Yes this is correct. I imagine that the original issue OP might have had the same problem. That would make this superseded by #13031 which succinctly describes this with no unknowns and gives a workaround. Going to close this as duplicate of that. If you have a different issue with login redirection not related to `--secure=false`, please file a new issue.
